### PR TITLE
restore emplacement syntax (obsolete)

### PR DESCRIPTION
**Pull request by nikomatsakis**
Fix https://github.com/rust-lang/rust/issues/50832

r? @petrochenkov

### CODE REVIEW

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -2839,6 +2839,17 @@ impl<'a> Parser<'a> {
                 let (span, e) = self.interpolated_or_expr_span(e)?;
                 (lo.to(span), ExprKind::AddrOf(m, e))
             }
+            token::Ident(..) if self.token.is_keyword(keywords::In) => {
```
Comment (petrochenkov) [inline src/libsyntax/parse/parser.rs]: Wat.
I thought we removed the `in` version, like, years ago.

File: src/librustc_passes/ast_validation.rs [hunk 2]
```diff
@@ -172,6 +172,12 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
             ExprKind::InlineAsm(..) if !self.session.target.target.options.allow_asm => {
                 span_err!(self.session, expr.span, E0472, "asm! is unsupported on this target");
             }
+            ExprKind::ObsoleteInPlace(..) => {
+                self.err_handler()
+                    .struct_span_err(expr.span, "emplacement syntax is obsolete (for now, anyway)")
+                    .note("for more information, see <https://github.com/rust-lang/rust/issues/27779#issuecomment-378416911>")
```
Comment (petrochenkov) [inline src/librustc_passes/ast_validation.rs]: Tidy!

**Comment by petrochenkov** [comment]
r=me after fixing tidy

File: src/libsyntax/parse/parser.rs (hunk 1 above)
Comment (nikomatsakis) [reply src/libsyntax/parse/parser.rs]: Nope, it was removed in @aidanhs's PR

**Pull request merged**